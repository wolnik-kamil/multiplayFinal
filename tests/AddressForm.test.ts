import {describe, expect, it} from "vitest";
import {AddressForm} from "#components";
import {mount, VueWrapper} from "@vue/test-utils";
import {Multiselect} from "vue-multiselect";
import {mountSuspended} from "@nuxt/test-utils/runtime";


describe('inputs validation test', () => {
    test('zip code should be set like that: xx-xxx', async ()=> {
        const wrapper = mount(AddressForm)
        const inputField = wrapper.find<HTMLInputElement>('[data-test="input-zipCode"]');
        await inputField.setValue('12asdsdjak123');
        expect(inputField.element.value).toMatch('12-123');
    })
    test('multiselects are matching with snapshot', async () => {
        const wrapper = mountSuspended(AddressForm)
        const multiselect = (await wrapper).findComponent('[data-test="input-city"]')
        await multiselect.setValue('Knur√≥w')
        console.log(multiselect.element)
    })

})