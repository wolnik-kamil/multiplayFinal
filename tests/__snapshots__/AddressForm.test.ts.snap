// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`inputs validation test > typing in multiselect should refresh value of variable 1`] = `false`;
